EECS3311-PD1/src/main/java/UI/BottomMenuView.java

```diff
@@ -0,0 +1,195 @@
+package UI;
+
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.collections.FXCollections;
+import javafx.event.*;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.control.Button;
+import javafx.scene.control.ChoiceBox;
+import javafx.scene.layout.HBox;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BottomMenuView extends HBox {
+    @FXML
+    public ChoiceBox<String> availableCharts;
+    @FXML
+    public ChoiceBox<String> availableAnalyses;
+
+    @FXML
+    public Button addChartBtn;
+
+    @FXML
+    public Button removeChartBtn;
+
+    @FXML
+    public Button recalculateBtn;
+
+    public BottomMenuView() {
+        FXMLLoader fxmlLoader = new FXMLLoader(getClass().getResource(
+                "/fxml/BottomMenuView.fxml"));
+        fxmlLoader.setRoot(this);
+        fxmlLoader.setController(this);
+
+        try {
+            fxmlLoader.load();
+        } catch (IOException exception) {
+            throw new RuntimeException(exception);
+        }
+
+        addChartBtn.setOnAction(this::addChartToListOfViews);
+        removeChartBtn.setOnAction(this::removeChartFromListOfViews);
+        recalculateBtn.setOnAction(this::recalculate);
+
+        setAvailableCharts(new ArrayList<>());
+        setAvailableAnalyses(new ArrayList<>());
+    }
+
+    public void addChartToListOfViews(ActionEvent actionEvent) {
+        propertyOnChartAdded.get().handle(new SelectedChartEvent(availableCharts.getValue(), EventType.ROOT));
+    }
+
+    public void removeChartFromListOfViews(ActionEvent actionEvent) {
+        propertyOnChartRemoved.get().handle(new SelectedChartEvent(availableCharts.getValue(), EventType.ROOT));
+    }
+
+    public void recalculate(ActionEvent actionEvent) {
+        propertyOnRecalculatePressed.get().handle
+                (new RecalculateEvent(availableAnalyses.getValue(), availableCharts.getValue(), EventType.ROOT));
+    }
+
+    static public class SelectedChartEvent extends Event {
+        private String chartName;
+
+        public SelectedChartEvent(String chartName, EventType<? extends Event> eventType) {
+            super(eventType);
+            this.chartName = chartName;
+        }
+
+        public String getChartName() {
+            return chartName;
+        }
+    }
+
+    static public class RecalculateEvent extends Event {
+        private String analysisName;
+        private String chartName;
```

Comment:
Not sure if it makes sense to pass the data about the selected analysis and the selected chart, since we already have a way to subscribe to any changes in each one of those.

Note: if you end up with an event type that does not have any extra data, you can just use `ActionEvent`.